### DISCUSSION TITLE
.NET 7 service releases

### DISCUSSION THREAD

**Discussion by stevehansen** [dotnet/maui#13578] [General]
I see that we have our first .NET 8 preview last week but a lot of the bugs that have been fixed are blocking us at the moment for .NET 7, when can we expect updates on those versions? Do we get a release each month at the end (so around tomorrow?), as the last release was also the last day of January.

**Comment by nicjay** [comment]
According to the [.NET MAUI Support Policy](https://dotnet.microsoft.com/en-us/platform/support/policy/maui):

> Servicing updates are made available for supported versions of .NET MAUI. Servicing updates ship on the same monthly update cycle as the rest of .NET — in months where there are fixes, the update ships on the second Tuesday of the month (also known as Patch Tuesday) along with updates for .NET.
> 
> .NET MAUI updates may include a mix of security and non-security fixes in a single update.

However in reality, the services releases have been coming out at (seemingly) random times. It would be good to get some clarification on that.

**Comment by stevehansen** [comment]
so 2 .NET 8 preview releases, which only work with the CLI tools and not a single .NET 7 release yet since, is this what we have to expect?

**Reply by hartez** [reply]
.NET MAUI 7.0.59 Service Release 3 was shipped with VS 17.5 last month. 

Service Release 4 will likely ship next month, but those dates are subject to change.

**Reply by minze-it** [reply]
Excuse me, but SR3 was released in January. That's nearly two months ago.
If SR4 will be released in April, the pace is one service release every 3 months which is not really the expected "monthly update cycle".

**Reply by baaaaif** [reply]
It's really just annoying.
Since you don't stick to your own release cycles anyway - why wait until the next patch day now? It compiles, lets ship ;)
We are all eagerly awaiting...

Or at least adjust the page and bring in more clarity https://github.com/dotnet/maui/wiki/Roadmap#net-maui-7-servicing-updates @davidortinau

**Reply by hartez** [reply]
> Excuse me, but SR3 was released in January. That's nearly two months ago. If SR4 will be released in April, the pace is one service release every 3 months which is not really the expected "monthly update cycle".

SR3 was first included in VS 17.4.5, which was released February 14th: https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes-v17.4#1745--visual-studio-2022-version-1745

The SR3 release of MAUI was _tagged_ on January 31: https://github.com/dotnet/maui/releases/tag/7.0.59, is that what you're looking at? There's a lag time between tagging and release (for integration, testing, etc.).

**Reply by AndreasReitberger** [reply]
@hartez 
Wouldn't it be better if you shipped MAUI as nuget? So there is no need to wait for the Visual Studio releases anymore.
Also you could make nightly builds available with nugets.

**Reply by hartez** [reply]
> @hartez Wouldn't it be better if you shipped MAUI as nuget? So there is no need to wait for the Visual Studio releases anymore. Also you could make nightly builds available with nugets.

The decision to tie it to VS isn't something I have influence over. But there is work in progress to allow updating minor versions via nuget: https://github.com/dotnet/maui/pull/11206

So being able to have nightly builds is a possibility.

**Reply by AndreasReitberger** [reply]
@hartez 
It's good to see progress in that direction, thanks for sharing this information!

**Comment by minze-it** [comment]
It would be so wonderful to have at least some kind of info on the timeline... Really frustrating at the moment...

**Comment by LennoxP90** [comment]
where is the next SR? it has been 2 months...
